include/opt-sched/Scheduler/lnkd_lst.h

```diff
@@ -573,43 +573,66 @@ inline T *PriorityList<T, K>::GetNxtPriorityElmnt(K &key) {
   }
 }
 
+//(Vlad) added functionality to decrease priority
+// used for decreasing priority of clusterable instrs
+// when leaving a cluster
 template <class T, class K>
 void PriorityList<T, K>::BoostEntry(KeyedEntry<T, K> *entry, K newKey) {
   KeyedEntry<T, K> *crnt;
   KeyedEntry<T, K> *next = entry->GetNext();
   KeyedEntry<T, K> *prev = entry->GetPrev();
 
-  assert(newKey > entry->key);
   assert(LinkedList<T>::topEntry_ != NULL);
 
-  entry->key = newKey;
+  if (entry->key < newKey) // behave normally
+  {
```

Comment:
Can you put the comment after the brace, or on the next line after the `if`? The LLVM code style places the `{` right after the `)` for the `if`, like so: `if (entry->key < newKey) {`, and it's not like that here because of the comment.

Reply:
Ok, I have updated the style. Thank you for pointing it out, old habits die hard.